vizro-core/docs/pages/user-guides/data.md

```diff
@@ -333,14 +336,77 @@ Parametrized data loading is compatible with [caching](#configure-cache). The ca
 
 You cannot pass [nested parameters](parameters.md#nested-parameters) to dynamic data. You can only target the top-level arguments of the data loading function, not the nested keys in a dictionary.
 
-### Filter update limitation
+### Filters
+
+By default, when a [filter](filters.md) depends on dynamic data, the available selector values update on page refresh to reflect the latest dynamic data. The mechanism for this, including caching, works exactly like other non-control components such as `vm.Graph`. However, unlike such components, a filter can depend on multiple data sources.
+
+Vizro looks at the data source for each component in the filter's `targets`. If at least one of these is a dynamic data source then the filter behaves dynamically and the filter's selector updates on page refresh. Remember that when `targets` is not explicitly specified, a filter applies to all the components on a page that use a DataFrame including `column`.
+
+When the page is refreshed, the behaviour of a dynamic filter is as follows:
```

Comment:
Nitpick, but I stumbled: what is a dynamic filter, it was never defined. Of course it is a filter behaving dynamically, but maybe we could be clearer in the language

Reply:
It might help to say clearly somwhere at the start: dynamic filter = no selector defined, or no options on selector. Static filter = options on selector defined?

This focusses less on the underlying data, but more on the filter. Maybe both can be described.

Reply:
I found this section a bit challenging to follow. Here's the explanation:

_If you have a filter that depends on dynamic data but do not want the available values to change when the dynamic data changes then you should manually specify the `selector`'s `options` field (categorical selector) or `min` and `max` fields (numerical selector)._

I appreciate how this is presented as a feature - it highlights the ability to maintain static options and filters even when working with dynamic data sources. However, I think it would be helpful to briefly mention this benefit earlier in the paragraph or include a link to the detailed explanation below for quicker reference.

Reply:
All very good points, thank you! Implemented some rewording and restructuring based on this.